src/scenic/syntax/compiler.py

```diff
@@ -171,3 +172,158 @@ def visit_ApparentlyFacingSpecifier(self, node: s.ApparentlyFacingSpecifier):
             if node.base is not None
             else [],
         )
+
+    # Operators
+
+    def visit_RelativePositionOp(self, node: s.RelativePositionOp):
+        return ast.Call(
+            func=ast.Name(id="RelativePosition", ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_RelativeHeadingOp(self, node: s.RelativeHeadingOp):
+        return ast.Call(
+            func=ast.Name(id="RelativeHeading", ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_ApparentHeadingOp(self, node: s.ApparentHeadingOp):
+        return ast.Call(
+            func=ast.Name(id="ApparentHeading", ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_DistanceFromOp(self, node: s.DistanceFromOp):
+        assert node.target is not None or node.base is not None
+        target = ego if node.target is None else self.visit(node.target)
+        base = ego if node.base is None else self.visit(node.base)
+        return ast.Call(
+            func=ast.Name(id="DistanceFrom", ctx=loadCtx),
+            args=[target],
+            keywords=[ast.keyword(arg="Y", value=base)],
+        )
+
+    def visit_DistancePastOp(self, node: s.DistancePastOp):
+        return ast.Call(
+            func=ast.Name(id="DistancePast", ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_AngleFromOp(self, node: s.AngleFromOp):
+        assert (
+            node.base is not None or node.target is not None
+        ), "neither target nor base were specified in AngleFromOp"
+        base = ego if node.base is None else self.visit(node.base)
+        target = ego if node.target is None else self.visit(node.target)
```

Comment:
As for `distance from`, for consistency maybe we should push the implicit ego behavior into the implementation in the veneer (although in this case you would have to change `AngleFrom` since it doesn't currently have an optional argument).

Reply:
I changed `AngleFrom` and made both arguments optional. If missing, `X` and `Y` will have `None` which will be replaced with `ego()`.

61e2e26f0471916125b9d34693bcd55470c02e78